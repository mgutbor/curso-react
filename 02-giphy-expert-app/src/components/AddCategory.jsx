import { useState } from "react";

export const AddCategory = ( { onNewCategory }) => {
  
  const [inputValue, setInputValue] = useState('')
  
  const handlerInputValue = ({target}) => {
    setInputValue(target.value)
  }

  const onSubmitForm = (event) => {
    event.preventDefault();
    const newInputValue = inputValue.trim();
    if (newInputValue.length <= 1) return;

    onNewCategory(newInputValue);
    setInputValue('');
  }

  return (
    <form onSubmit={ onSubmitForm}>
      <input 
          type="text"
          placeholder="Buscar GIFS"
          value={ inputValue }
          onChange={ handlerInputValue }
      />
    </form>
  )
}
