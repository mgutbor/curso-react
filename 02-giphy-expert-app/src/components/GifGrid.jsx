import { useState, useEffect } from "react";
import { getGif } from "../helpers/getGifs";

export const GifGrid = ( { category } ) => {

  const [images, setImages] = useState([])

  useEffect( () => {
    getGif(category)
      .then( newImages => setImages(newImages));
  }, []);

  return (
    <>
      <h3>{ category }</h3>
      <ol>
      { images.map( ({id, title}) => {
          return <li key={ id }>{ title }</li>
        })
      }
      </ol>
    </>
  )
}
