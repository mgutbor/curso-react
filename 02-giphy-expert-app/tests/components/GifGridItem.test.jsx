import { render } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';


describe('Pruebas en GifGridItem', () => {
  
  const title = 'Manolo';
  const url = 'https://manolo.com'

  test('debe hacer match con el snapshot', () => {
  
    const { container } = render( <GifGridItem title={ title } url={ url } /> );

    expect( container ).toMatchSnapshot();
    
  });

});