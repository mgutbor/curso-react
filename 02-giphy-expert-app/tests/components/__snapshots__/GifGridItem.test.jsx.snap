// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Manolo"
      src="https://manolo.com"
    />
    <p>
      Manolo
    </p>
  </div>
</div>
`;
